clippy_lints/src/methods/mod.rs

```diff
@@ -1670,6 +1676,48 @@ impl<'tcx> LateLintPass<'tcx> for Methods {
             }
         }
     }
+
+    fn check_trait_item(&mut self, cx: &LateContext<'tcx>, item: &'tcx TraitItem<'_>) {
```

Comment:
We could add an example in the lint's description showing how it works with traits. 

Could you also please fix the existing examples by removing the indentation before the `#` symbols? In the [webpage](https://rust-lang.github.io/rust-clippy/master/#new_ret_no_self) it's showing the return type for the last two cases, when it shouldn't.